### PR TITLE
Two stocks comparison

### PR DESCRIPTION
**Pull request by murradkh**
changes include in this PR:
1. moving the embedded JavaScript code into HTML to a separated file in `myapp/static/js/myapp/single_stock.js`
2. adding the functionality for retrieving/listing stocks symbols & names, when typing any key in the input field.
3. adding UI for viewing the list of stocks names and choosing the desired stock to compare with.
4. editing canvas js for integrating a new chart Into the original chart.

Screenshot 2020-10-29 011137
Screenshot 2020-10-29 011222
Screenshot 2020-10-29 011204

### CODE REVIEW

File: myapp/static/js/myapp/single_stock.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            data: {
+                labels: historic_data_1.map(d => d.label),
+                datasets: datasets
+            },
+            options: {
+                responsive: true
+            }
+        });
+    });
+};
```
Comment (kamalq97) [inline myapp/static/js/myapp/single_stock.js:53]: Make sure this has all the newer changes and fixes I made in PR #32

File: myapp/stock_api.py [hunk 2]
```diff
@@ -68,9 +68,14 @@ def get_stock_info(symbol):
         raise e
 
 
-def get_stock_historic_prices(symbol, time_range='1m'):
+def get_stock_historic_prices(symbols, time_range='1m'):
     try:
-        return _request_data('/stable/stock/{symbol}/chart/{time_range}'.format(symbol=symbol, time_range=time_range))
+        # return _request_data('/stable/stock/{symbol}/chart/{time_range}'.format(symbol=symbol, time_range=time_range))
+        response = _request_data('/stable/stock/market/batch?symbols={symbols}&types=chart&range={time_range}'.format(
```
Comment (kamalq97) [inline myapp/stock_api.py:74]: I see you are now using the batch operation I told you about, Nice!!!

File: myapp/templates/single_stock.html [hunk 3]
```diff
@@ -33,124 +41,99 @@
...
+            <li class="nav-item">
+            <a class="nav-link" data-toggle="pill" href="#6m" role="tab" aria-controls="pills-contact" aria-selected="false">6M</a>
+          </li>
+            <li class="nav-item">
+            <a class="nav-link" data-toggle="pill" href="#1y" role="tab" aria-controls="pills-contact" aria-selected="false">1Y</a>
+          </li>
+            <li class="nav-item">
+            <a class="nav-link" data-toggle="pill" href="#max" role="tab" aria-controls="pills-contact" aria-selected="false">Max</a>
+          </li>
+        </ul>
```
Comment (kamalq97) [inline myapp/templates/single_stock.html:106]: A lot of work is needed here to ensure there are no merge conflicts between this and PR #32

File: myapp/templates/base.html [hunk 4]
```diff
@@ -15,6 +15,7 @@
     <link rel="stylesheet" href="{% static 'css/font-awesome.min.css' %}">
     <link href="{% static 'css/addons/datatables.min.css' %}" rel="stylesheet">
     <link href="{% static 'css/main.css' %}" rel="stylesheet" type="text/css">
+      <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
```
Comment (kamalq97) [inline myapp/templates/base.html]: Not sure why you have to import another icon font; Font Awesome 'fa fa-' works fine

**Review by kamalq97 — COMMENTED** [review]
Excellent work! The ability to make comparisons of more than two stocks and the ability to add stocks not in the database can be added in separate PRs.

File: myapp/templates/single_stock.html (hunk 3 above)
Comment (murradkh) [reply myapp/templates/single_stock.html:106]: yeah, I will do this after you merge your PR.

**Comment by murradkh** [comment]
> Excellent work! The ability to make comparisons of more than two stocks and the ability to add stocks, not in the database can be added in separate PRs.

thanks,

File: myapp/templates/base.html (hunk 4 above)
Comment (murradkh) [reply myapp/templates/base.html]: good to know

**Review by kamalq97 — APPROVED** [review]
Excellent work! I think merging this pull request into the master branch is going to take a lot of effort, especially since PR #32 made a lot of changes in the same area. I am happy to help you with merging this PR and ensuring that the 'comparison' and 'time range' features work well together

**Pull request closed without merging**